### PR TITLE
Organize MenuDrawer Adapter's code

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1989

### CODE REVIEW

File: gradle/libs.versions.toml
Comment (LunarX) [inline gradle/libs.versions.toml:1]: I think you commited one too many files

File: app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt [hunk 1]
```diff
@@ -243,59 +246,59 @@ class MenuDrawerAdapter @Inject constructor() : ListAdapter<Any, MenuDrawerViewH
...
-        MAILBOXES_HEADER_CLICKED,
+    enum class ViewType {
+        DIVIDER_VIEW,
+        MAILBOXES_HEADER_VIEW,
+        MAILBOX_VIEW,
+        INVALID_MAILBOX_VIEW,
+        FOLDERS_HEADER_VIEW,
+        FOLDER_VIEW,
+        EMPTY_FOLDERS_VIEW,
+        FOOTER_VIEW,
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt:262]: Did you try making ItemType and ViewType a single enum?

File: app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt [hunk 2]
```diff
@@ -243,59 +246,59 @@ class MenuDrawerAdapter @Inject constructor() : ListAdapter<Any, MenuDrawerViewH
     abstract class MenuDrawerViewHolder(val binding: ViewBinding) : ViewHolder(binding.root)
 
     private enum class ItemType {
-        CUSTOM_FOLDERS_HEADER,
-        EMPTY_CUSTOM_FOLDERS,
-        DIVIDER,
+        DIVIDER_ITEM,
+        FOLDERS_HEADER_ITEM,
+        EMPTY_CUSTOM_FOLDERS_ITEM,
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt:251]: You've left the term "custom" in the enum, was it the one you kept removing from the methods in the other files?

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/menuDrawer/MenuDrawerAdapter.kt]: Done.

**Review by LunarX — APPROVED** [review]

**Pull request merged**